packages/node-patches/src/fs.ts

```diff
@@ -0,0 +1,403 @@
+      );
+      //tslint:disable-next-line:no-any
+      (e as any).code = 'EINVAL';
+      throw e;
+    }
+    return str;
+  };
+
+  //tslint:disable-next-line:no-any
+  fs.readdir = (...args: any[]) => {
+    const p = path.resolve(args[0]);
+
+    let cb = args[args.length - 1];
+    if (typeof cb !== 'function') {
+      // this will likely throw callback required error.
+      return origReaddir(...args);
+    }
+
+    cb = once(cb);
+    args[args.length - 1] = (err: Error, result: Dirent[]) => {
+      if (err) return cb(err);
+      // user requested withFileTypes
+      if (result[0] && result[0].isSymbolicLink) {
+        Promise.all(result.map((v: Dirent) => handleDirent(p, v)))
+          .then(() => {
+            cb(null, result);
+          })
+          .catch(err => {
+            cb(err);
+          });
+      } else {
+        // string array return for readdir.
+        cb(null, result);
+      }
+    };
+
+    origReaddir(...args);
+  };
+
+  //tslint:disable-next-line:no-any
+  fs.readdirSync = (...args: any[]) => {
+    const res = origReaddirSync(...args);
+    const p = path.resolve(args[0]);
+    //tslint:disable-next-line:no-any
+    res.forEach((v: Dirent | any) => {
+      handleDirentSync(p, v);
+    });
+    return res;
+  };
+
+  // i need to use this twice in bodt readdor and readdirSync. maybe in fs.Dir
+  //tslint:disable-next-line:no-any
+  function patchDirent(dirent: Dirent | any, stat: Stats | any) {
+    // add all stat is methods to Dirent instances with their result.
+    for (const i in stat) {
+      if (i.indexOf('is') === 0 && typeof stat[i] === 'function') {
+        //
+        const result = stat[i]();
+        if (result) dirent[i] = () => true;
+        else dirent[i] = () => false;
+      }
+    }
+  }
+
+  if (fs.opendir) {
+    const origOpendir = fs.opendir.bind(fs);
+    //tslint:disable-next-line:no-any
+    fs.opendir = (...args: any[]) => {
+      let cb = args[args.length - 1];
+      // if this is not a function opendir should throw an error.
+      //we call it so we don't have to throw a mock
+      if (typeof cb === 'function') {
+        cb = once(cb);
+        args[args.length - 1] = async (err: Error, dir: Dir) => {
+          try {
+            cb(null, await handleDir(dir));
+          } catch (e) {
+            cb(e);
+          }
+        };
+        origOpendir(...args);
+      } else {
+        return origOpendir(...args).then((dir: Dir) => {
+          return handleDir(dir);
+        });
+      }
+    };
+  }
+
+  async function handleDir(dir: Dir) {
+    const p = path.resolve(dir.path);
+    const origIterator = dir[Symbol.asyncIterator].bind(dir);
+    //tslint:disable-next-line:no-any
+    const origRead: any = dir.read.bind(dir);
+
+    dir[Symbol.asyncIterator] = async function*() {
+      for await (const entry of origIterator()) {
+        await handleDirent(p, entry);
+        yield entry;
+      }
+    };
+
+    //tslint:disable-next-line:no-any
```

Comment:
you could fix most of these `any` by replacing them with `unknown` where appropriate

Reply:
i was torn because like the core types have any's and these types don't leak to api callers because this mutates an fs at runtime. its worth a shot just to clean up these comments though =)

Reply:
i tried and it looks cool but i'm not good enough at it clearly because it makes everything red underline